src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -344,6 +345,7 @@ class TypeLabelController<T> implements IDisposable {
 
 	private automaticKeyboardNavigation = true;
 	private triggered = false;
+	private charactersTyped = 0;
```

Comment:
Shouldn't we rather be storing the previously declared item's label? Or the element index? Then you could compare labels/index and if it hasn't changed, then we don't announce it, right?

Reply:
@joaomoreno makes sense, I have pushed that change now. Check it out. I tested and works nice.